packages/material/test/renderers/MaterialOneOfRenderer.test.tsx

```diff
@@ -0,0 +1,404 @@
+    wrapper = mount(
+      <Provider store={store}>
+        <MaterialOneOfRenderer schema={schema} uischema={uischema} />
+      </Provider>
+    );
+    const input = wrapper.find('input').first();
+    input.simulate('change', { target: { value: 'test' }});
+    wrapper.update();
+    expect(store.getState().jsonforms.core.data).toEqual({
+      value: 'test'
+    });
+  });
+
+  it('should add an item within an array', () => {
+    const schema = {
+      type: 'object',
+      properties: {
+        oneOrMoreThings: {
+          oneOf: [
+            {
+              $ref: '#/definitions/thing'
+            },
+            {
+              $ref: '#/definitions/thingArray'
+            }
+          ]
+        }
+      },
+      definitions: {
+        thing: {
+          title: 'Thing',
+          type: 'string'
+        },
+        thingArray: {
+          title: 'Things',
+          type: 'array',
+          items: {
+            $ref: '#/definitions/thing'
+          }
+        }
+      }
+    };
+    const resolvedSchema = {
+      type: 'object',
+      properties: {
+        thingOrThings: {
+          oneOf: [
+            {
+              title: 'Thing',
+              type: 'string',
+            },
+            {
+              title: 'Things',
+              type: 'array',
+              items: {
+                title: 'Thing',
+                type: 'string'
+              }
+            }
+          ]
+        }
+      }
+    };
+    const uischema: ControlElement = {
+      type: 'Control',
+      scope: '#/properties/thingOrThings'
+    };
+
+    const store = initStore();
+    store.dispatch(Actions.init({ data: {}, schema, uischema}));
+    const promise = Promise.resolve({ resolved: resolvedSchema});
+    const mock = jest.spyOn(JsonRefs, 'resolveRefs');
+    mock.mockReturnValue(promise);
+
+    wrapper = mount(
+      <Provider store={store}>
+        <JsonForms schema={schema} uischema={uischema} />
+      </Provider>
+    );
+
+    return promise.then(() => {
+      // expect(wrapper.state())
+      wrapper.update();
+
+      // select oneOf-1
+      let buttons = wrapper.find('button');
+      buttons.at(1).simulate('click');
+      wrapper.update();
+
+      // confirm dialog
+      buttons = wrapper.find('button');
+      buttons.at(3).simulate('click');
+      wrapper.update();
+
+      // click add button
+      buttons = wrapper.find('button');
+      const addButton = buttons.at(2);
+      const nrOfRowsBeforeAdd = wrapper.find('tr');
+      addButton.simulate('click');
+      addButton.simulate('click');
+
+      // check if row has been added
+      wrapper.update();
+      const nrOfRowsAfterAdd = wrapper.find('tr');
+
+      // 1 header row + no data row
+      expect(nrOfRowsBeforeAdd.length).toBe(2);
+      // 1 header row + 2 data rows (one is replacing the 'No data' one)
+      expect(nrOfRowsAfterAdd.length).toBe(3);
+    });
+  });
+
+  it('should add an object within an array', () => {
```

Comment:
isn't this exactly the same as the previous except one more assert?

Reply:
Yes, it's only the schema that differs.

Reply:
I now see the different schemas, but I just tried out using one schema for both tests and they still work. The array part which is the only one being checked is the same for both schemas.